src/test/java/com/blog/portal/serviceimpl/ReactionServiceImplTest.java

```diff
@@ -0,0 +1,141 @@
+package com.blog.portal.serviceimpl;
+
+import com.blog.portal.entities.Blog;
+import com.blog.portal.entities.Reaction;
+import com.blog.portal.enumResource.React;
+import com.blog.portal.exception.ResourceNotFoundException;
+import com.blog.portal.mapper.ReactOnBlogMapper;
+import com.blog.portal.repository.BlogRepository;
+import com.blog.portal.repository.ReactionRepository;
+import com.blog.portal.requestPayload.ReactOnBlogInDto;
+import com.blog.portal.responsePayload.ReactionOnBlogOutDto;
+import com.blog.portal.services.ReactionService;
+import com.blog.portal.serviceimpl.ReactionServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+public class ReactionServiceImplTest {
+
+    @Mock
+    private ReactionRepository likeAndDislikeRepo;
+
+    @Mock
+    private BlogRepository blogRepository;
+
+    @InjectMocks
+    private ReactionService reactionService = new ReactionServiceImpl();
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    public void testDoReactOnPost_ReactLike_NewReaction() {
+        ReactOnBlogInDto inDto = new ReactOnBlogInDto("userId", "postId", React.LIKE);
+        Reaction userReaction = null;
+        Blog blog = createSamplePost();
+        when(likeAndDislikeRepo.findByUserIdAndPostId("userId", "postId")).thenReturn(userReaction);
+        when(blogRepository.findById("postId")).thenReturn(Optional.of(blog));
+        when(blogRepository.save(blog)).thenReturn(blog);
+        when(likeAndDislikeRepo.save(any(Reaction.class))).thenAnswer(invocation -> invocation.getArgument(0));
+
+        ReactionOnBlogOutDto result = reactionService.doReactOnBlog(inDto);
+
+        assertNotNull(result);
+        assertEquals(React.LIKE, result.getType());
+        assertTrue(blog.getLikedBy().contains("userId"));
+        assertFalse(blog.getDisLikedBy().contains("userId"));
```

Comment:
no need to check notNull and fields individually, check whole object. expected and actual